src/node_lkl.cc

```diff
@@ -150,3 +148,39 @@ NAN_METHOD(syscall) {
 		}
 	}
 }
+
+NAN_METHOD(parseDirent64) {
+	// Args:
+	// * Buffer buffer
+	// * int nread
+	// * Array result
+	// Parses buffer (which should be a lkl_linux_dirent64*) and fills the
+	// result array with filenames it finds.
+	// Filenames are Buffers that need to be decoded by the caller.
+	char* buf = (char*) node::Buffer::Data(info[0]);
+	unsigned int nread = info[1]->Uint32Value();
+	assert(nread <= node::Buffer::Length(info[0]));
```

Comment:
Here we trust the caller to provide a correct `nread` value assert that it's less than the length. Since buffers carry a `(char * buf, int len)` pair in them, would it make sense to make this function accept a single buffer? In fact this implementation can cause problems because the buffer passed to `lkl_getdents64` is a statically allocated 1024 byte -sized buffer than can contain non-zero memory. So even though `nread` will be less than the buffer's length, the data within it might result in memory problems, for example attempting to ro `strlen` at a memory address that doesn't have a null terminated string.